autoload/sj/r.vim

```diff
@@ -0,0 +1,190 @@
+  let items = map(items, {k, v -> v . (k+1 < len(items) ? "," : "")})
+
+  if g:r_indent_align_args && len(items)
+    let items[0]  = "(" . items[0]
+    let items[-1] = items[-1] . ")"
+    let lines = items
+  else
+    let lines = ["("] + items + [")"]
+  endif
+
+  call sj#PopCursor()
+  call s:ReplaceMotionPreserveCursor('va(', lines)
+
+  return 1
+endfunction
+
+" function! sj#r#JoinFuncall()
+"
+" Join an R function call if the cursor lies within the arguments of a
+" function call
+"
+function! sj#r#JoinFuncall()
+  if !s:IsValidSelection("va(")
+    return 0
+  endif
+
+  call sj#PushCursor()
+  let items = s:ParseJsonFromMotion("va(\<esc>vi(")
+
+  " clean up unwanted spaces around parens which can occur during nested joins
+  let text = join(items, ", ")
+  let text = substitute(text, '(\s', '(', 'g')
+  let text = substitute(text, '\s)', ')', 'g')
+
+  call sj#PopCursor()
+  call s:ReplaceMotionPreserveCursor("va(", ["(" . text . ")"])
+
+  return 1
+endfunction
+
+" function! s:Triml(text [, mask])
+"
+" A shorthand for trim with setting dir=1
+"
+function! s:Triml(text, ...)
+  let inlen = len(a:text)
+  let text = a:0 > 0
+        \ ? trim(a:text, a:1, 1)
+        \ : substitute(a:text, '^\s*', '', 'g')
+  return [text, inlen - len(text)]
+endfunction
+
+" function! s:MoveCursor(lines, cols)
+"
+" Reposition cursor given relative lines offset and columns from the start of
+" the line
+"
+function! s:MoveCursor(lines, cols)
+  let y = a:lines > 0 ? a:lines . 'j^' : a:lines < 0 ? a:lines . 'k^' : ''
+  let x = a:cols  > 0 ? a:cols  . 'l'  : a:cols  < 0 ? a:cols  . 'h'  : ''
+  let motion = y . x
+  if len(motion)
+    execute 'silent normal! ' . motion
+  endif
+endfunction
+
+" function! s:ParseJsonObject(text)
+"
+" Wrapper around sj#argparser#js#Construct to simply parse a given string
+"
+function! s:ParseJsonObject(text)
+  let parser = sj#argparser#js#Construct(0, len(a:text), a:text)
+  call parser.Process()
+  return parser.args
+endfunction
+
+" function! sj#r#ParseFromMotion(motion)
+"
+" Parse a json object from the visual selection of a given normal-mode motion
+" string
+"
+function! s:ParseJsonFromMotion(motion)
+  let text = sj#GetMotion(a:motion)
+  return s:ParseJsonObject(text)
+endfunction
+
+" function! s:GetTextRange(start, end)
+"
+" Get the text between positions marked by getpos("start") and getpos("end")
+function! s:GetTextRange(start, end)
+  let text = sj#GetByPosition(getpos(a:start), getpos(a:end))
+  let lines = split(text, "\n")
+
+  return lines
+endfunction
+
+" function! s:IsValidSelection(motion)
+"
+" Test whether a visual selection contains more than a single character after
+" performing the given normal-mode motion string
+"
+function! s:IsValidSelection(motion)
+  call sj#PushCursor()
+  execute "silent normal! " . a:motion . "\<esc>"
+  execute "silent normal! \<esc>"
+  let is_valid = getpos("'<") != getpos("'>")
+  call sj#PopCursor()
+  return is_valid
+endfunction
```

Comment:
This code could be reduced to `return sj#GetMotion(a:motion) != ''`, but it would be slightly more expensive, given it allocates strings and such. I don't think it'd be a big deal, but I've left it as it is, just in case.

Reply:
Since this is used to determine when a split/join should fail, I think it's important to keep it fast. I don't have a good sense for vimscript's performance considerations, so I'll lean on your judgement to make a call about which method you prefer.

Reply:
Let's leave it like this for now.